pulp_container/app/tasks/sync_stages.py

```diff
@@ -117,18 +100,8 @@ async def get_signature_source(self):
             message="Downloading tag list", code="sync.downloading.tag_list", total=1
         ) as pb:
             repo_name = self.remote.namespaced_upstream_name
-            relative_url = "/v2/{name}/tags/list".format(name=repo_name)
-            tag_list_url = urljoin(self.remote.url, relative_url)
-            list_downloader = self.remote.get_downloader(url=tag_list_url)
-            await list_downloader.run(extra_data={"repo_name": repo_name})
-
-            with open(list_downloader.path) as tags_raw:
-                tags_dict = json.loads(tags_raw.read())
-                tag_list = tags_dict["tags"]
-
-            # check for the presence of the pagination link header
-            link = list_downloader.response_headers.get("Link")
-            await self.handle_pagination(link, repo_name, tag_list)
+            tag_list_url = "/v2/{name}/tags/list".format(name=repo_name)
+            tag_list = await self.get_paginated_tag_list(tag_list_url, repo_name)
```

Comment:
i am afraid this will not work for registries that do not have  pagination

Reply:
ah nevermind, we just do extra autopsy to the `tag_list_url`

Reply:
I still changed the code flow a little more to skip the rel_path dance on the first `link=tag_list_url` round.

Reply:
yes thank you :+1: